[12.0][IMP] Add a default Key to the API Key model

danypr92 opened the pull request:
This MR generates a random UUID4 value as a key for new API Key instances.
The value is generated with the  uuid4 random generator of UUID package:

https://docs.python.org/3.8/library/uuid.html#uuid.uuid4

Code moved from 'easy_my_coop_api' module of the repository 'coopiteasy/vertical-cooperative':

https://github.com/coopiteasy/vertical-cooperative/blob/12.0/easy_my_coop_api/models/auth_api_key.py#L20

We propose to move this default value to the main `auth_api_key` module instead of having it in`easy_my_coop_api` because we think that this feature can help every user of this module.

danypr92 committed:
Add a default Key to the API Key model

The value is generated with the  uuid4 random generator of uuid package:

https://docs.python.org/3.8/library/uuid.html#uuid.uuid4

Code moved from 'coopiteasy/vertical-cooperative' from the
'easy_my_coop_api' module:

https://github.com/coopiteasy/vertical-cooperative/blob/12.0/easy_my_coop_api/models/auth_api_key.py#L20

danypr92 commented:
@lmignon @rousseldenis Can you review this proposal? 
Thanks!

lmignon commented on auth_api_key/models/auth_api_key.py:
```diff
@@ -13,10 +14,14 @@ class AuthApiKey(models.Model):
     key = fields.Char(
         required=True,
-        help="""The API key. Enter a dummy value in this field if it is
+        default=_default_key,
```
@danypr92 To allow an easy override of the default method, you should avoid to make a direct link to the method into the field definition. `default=lambda self: self._default_key()`

danypr92 replied:
Thanks @lmignon !!

I push a commit with your suggestion. :rocket:

lmignon reviewed (changes requested):
@danypr92 Thank you for your contribution. A little comment on my side, otherwise LGTM

danypr92 committed:
Avoid to make a direct link to the method into the field definition

This change allow an easy override of the default method.

Suggested by @lmignon

lmignon reviewed (approved):
Thank you for your contribution! LGTM (Code review)

lmignon commented:
ping @rousseldenis

danypr92 commented:
@rousseldenis, any news?

danypr92 commented:
Thanks a lot @rousseldenis and @lmignon!

:heart_eyes: :heart_eyes: :heart_eyes:

lmignon commented:
/ocabot merge patch
